### PR TITLE
#7034 NPE when setting preferred size and null chat, in ChatPlayerPan…

### PR DESCRIPTION
**Pull request by tjrbrom**
…el#setDynamicPreferredSize

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[] Problem fix
[] Other:

### CODE REVIEW

**Comment by DanVanAtta** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- game-core/src/test/java/games/strategy/engine/chat/ChatPlayerPanelTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5827688&bid=19080972)

**Comment by RoiEXLab** [comment]
FYI there's no need to close and open a new PR every time you make a change

**Comment by tjrbrom** [comment]
> FYI there's no need to close and open a new PR every time you make a change

I'm sorry I didn't mean to, still learning how to properly sync between forks

File: game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java [hunk 1]
```diff
@@ -85,9 +85,11 @@ public void setChat(final Chat chat) {
    */
   private void setDynamicPreferredSize() {
     int maxNameLength = 0;
-    final FontMetrics fontMetrics = this.getFontMetrics(UIManager.getFont("TextField.font"));
-    for (final UserName onlinePlayer : chat.getOnlinePlayers()) {
-      maxNameLength = Math.max(maxNameLength, fontMetrics.stringWidth(onlinePlayer.getValue()));
+    if (chat != null) {
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java:88]: I was thinking about this for a while, but I think this isn't the right fix for this.
The stacktrace in #7034 makes it clear that this `NullPointerException` occurs when shutting down a game, so adjusting the preffered size just before the game is closed doesn't really make sense to me.

That's why I'd recommend making this method no-op if `chat` is null i.e.
```java
if (chat == null) {
  return;
}
```
At the start if the method

**Comment by DanVanAtta** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- game-core/src/test/java/games/strategy/engine/chat/ChatPlayerPanelTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5827688&bid=19080972)

File: game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java (hunk 1 above)
Comment (tjrbrom) [reply game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: Just pushed that, and verifying zero interactions with renderer mock. Was wondering if I should create more unit tests/test cases (in general) ?

**Comment by DanVanAtta** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- game-core/src/test/java/games/strategy/engine/chat/ChatPlayerPanelTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5827688&bid=19080972)

**Pull request merged**